src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -212,53 +213,103 @@ public void validateIntegrationTestClassNames() throws Exception {
 
     @Test
     public void validateInputFiles() throws Exception {
-        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir)) {
+        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(
+                inputDir,
+                path -> {
+                    return path.toFile().isDirectory();
+                }
+        )) {
             for (Path dir : dirs) {
-                // input directory must be named in lower case
-                assertWithMessage(dir + " is not a directory")
-                        .that(Files.isDirectory(dir))
-                        .isTrue();
-                final String dirName = dir.toFile().getName();
-                assertWithMessage("Invalid directory name: " + dirName)
-                        .that(ALLOWED_DIRECTORY_AND_CHECKS)
-                        .containsKey(dirName);
-
-                // input directory must be connected to an existing test
-                final String check = ALLOWED_DIRECTORY_AND_CHECKS.get(dirName);
-                final Path javaPath = javaDir.resolve("XpathRegression" + check + "Test.java");
-                assertWithMessage("Input directory '" + dir
-                            + "' is not connected to Java test case: " + javaPath)
-                        .that(Files.exists(javaPath))
-                        .isTrue();
-
-                // input files should be named correctly
-                validateInputDirectory(dir);
+                validateInputFiles(dir, dir.getFileName().toString());
             }
         }
     }
 
-    private static void validateInputDirectory(Path checkDir) throws IOException {
-        final Pattern pattern = Pattern.compile("^InputXpath(.+)\\.java$");
-        final String check = ALLOWED_DIRECTORY_AND_CHECKS.get(checkDir.toFile().getName());
-
-        try (DirectoryStream<Path> inputPaths = Files.newDirectoryStream(checkDir)) {
-            for (Path inputPath : inputPaths) {
-                final String filename = inputPath.toFile().getName();
-                if (filename.endsWith("java")) {
```

Comment:
`.java` was although correct , right?

Reply:
this is not really good assumption, but we should keep it, until we make support for config to be placed in other file types, for now our test suite support only java.

Reply:
If we only support `java` then `.java` is okay, as its equivalent as before `like` java but more specific matching for a file and not a file ending with java. I would like to use the already used `.java` so we can centralise the the value.